tests/test_container_tasks.py

```diff
@@ -9,64 +8,45 @@
 from conda_forge_tick.update_upstream_versions import (
     all_version_sources,
     get_latest_version_containerized,
 )
+from conda_forge_tick.utils import run_container_task
 
 
-def test_container_tasks_version():
-    res = subprocess.run(
-        [
-            "docker",
-            "run",
-            "--rm",
-            "-t",
-            "conda-forge-tick:test",
-            "python",
-            "/opt/autotick-bot/docker/run_bot_task.py",
-            "update-version",
-            "--existing-feedstock-node-attrs=conda-smithy",
-        ],
-        capture_output=True,
+def test_container_tasks_get_latest_version():
+    data = run_container_task(
+        "get-latest-version",
+        ["--existing-feedstock-node-attrs", "conda-smithy"],
     )
-    assert res.returncode == 0
-    data = json.loads(res.stdout.decode("utf-8"))
     assert data["new_version"] == conda_smithy.__version__
 
 
-def test_container_tasks_version_json():
-    with lazy_json_override_backends(["github"], use_file_cache=False):
-        with LazyJson("node_attrs/conda-smithy.json") as lzj:
-            existing_feedstock_node_attrs = dumps(lzj.data)
+def test_container_tasks_get_latest_version_json():
+    with tempfile.TemporaryDirectory() as tmpdir, pushd(tmpdir):
+        with lazy_json_override_backends(["github"], use_file_cache=False):
+            with LazyJson("node_attrs/conda-smithy.json") as lzj:
+                existing_feedstock_node_attrs = dumps(lzj.data)
 
-    res = subprocess.run(
-        [
-            "docker",
-            "run",
-            "--rm",
-            "-t",
-            "conda-forge-tick:test",
-            "python",
-            "/opt/autotick-bot/docker/run_bot_task.py",
-            "update-version",
-            "--existing-feedstock-node-attrs",
-            existing_feedstock_node_attrs,
-        ],
-        capture_output=True,
-    )
-    assert res.returncode == 0
-    data = json.loads(res.stdout.decode("utf-8"))
-    assert data["new_version"] == conda_smithy.__version__
+        data = run_container_task(
+            "get-latest-version",
+            [
+                "--existing-feedstock-node-attrs",
+                existing_feedstock_node_attrs,
+            ],
+        )
+        assert data["new_version"] == conda_smithy.__version__
 
 
 def test_get_latest_version_containerized():
-    # if the user doesn't set CI, assume we are in CI
-    with lazy_json_override_backends(["github"], use_file_cache=False):
-        with LazyJson("node_attrs/conda-smithy.json") as lzj:
-            attrs = copy.deepcopy(lzj.data)
-
-    data = get_latest_version_containerized(
-        "conda-smithy", attrs, all_version_sources()
-    )
-    assert data["new_version"] == conda_smithy.__version__
+    with tempfile.TemporaryDirectory() as tmpdir, pushd(tmpdir):
+        with lazy_json_override_backends(["github"], use_file_cache=False):
+            with LazyJson("node_attrs/conda-smithy.json") as lzj:
```

Comment:
If we require Python 3.10 for the bot code, then we can use [parenthesised context managers](https://docs.python.org/3/whatsnew/3.10.html#parenthesized-context-managers) for cleaner blocks:

```
with (
    tempfile.TemporaryDirectory() as tmpdir,
    pushd(tmpdir),
    lazy_json_override_backends(["github"], use_file_cache=False),
    LazyJson("node_attrs/conda-smithy.json") as lzj,
):
```